Gentoo specific file locations and commands

Patch by Gordon Bos

--- client.cpp
+++ client.cpp
@@ -20,6 +20,8 @@
 #include <string>
 #include <iostream>
 #include <libeutils/json/json.h>
+#include <unistd.h>
+#include <sys/types.h>
 
 int main( int argc, char** argv ) {
 
@@ -32,8 +34,8 @@
 		std::cerr << "Requires a command" << std::endl;
 		return 1;
 	}
-	std::string command("/usr/sbin/bubba-networkmanager --socket /tmp/bubba-networkmanager.sock --config /etc/bubba-networkmanager.conf");
-	std::string socket("/tmp/bubba-networkmanager.sock");
+	std::string command("/opt/bubba/sbin/bubba-networkmanager --socket /run/bubba-networkmanager.sock --config /etc/bubba/networkmanager.conf");
+	std::string socket("/run/bubba-networkmanager.sock");
 	EUtils::NetClient client( command, socket );
 	Json::Reader reader;
 	Json::FastWriter writer;
--- controllers/InterfaceController.cpp
+++ controllers/InterfaceController.cpp
@@ -632,11 +632,11 @@
 }
 
 bool InterfaceController::Up(const string& ifname){
-	return do_call("/sbin/ifup "+ifname+" 2>/dev/null")==0;
+	return do_call("initd=$(/sbin/rc-service -ir net."+ifname+") && ${initd} start")==0;
 }
 
 bool InterfaceController::Down(const string& ifname){
-	return do_call("/sbin/ifdown "+ifname+" 2>/dev/null")==0;
+	return do_call("initd=$(/sbin/rc-service -ir net."+ifname+") && ${initd} -D stop")==0;
 }

 
--- examplecfg/nmconfig
+++ examplecfg/nmconfig
@@ -1,5 +1,5 @@
 # Internal config file for bubba-network manager. DO not edit
 wanif=eth0
 lanif=eth1
-tz_map_file=/usr/share/bubba-networkmanager/tz-lc.txt
+tz_map_file=/var/lib/bubba/tz-lc.txt
 tz_file=/etc/timezone
--- main.cpp
+++ main.cpp
@@ -56,9 +56,9 @@
     p.AddOption( Option( "debug",'d',Option::Int,"Set debug level","value 0-7 (default is 5 and 7 is max)","5" ) );
     p.AddOption( Option( "version",'v',Option::None,"Show version","","false" ) );
     p.AddOption( Option( "ttl",'t',Option::Int,"time to live in seconds","60","60" ) );
-    p.AddOption( Option( "pidfile", '\0', Option::String, "PID file", "/var/run/bubba-networkmanager.pid", "/var/run/bubba-networkmanager.pid" ) );
-    p.AddOption( Option( "socket", '\0', Option::String, "Socket to communicate on", "/tmp/bubba-networkmanager.sock", "/tmp/bubba-networkmanager.sock" ) );
-    p.AddOption( Option( "config", '\0', Option::String, "Configuration file to read from", "/etc/bubba-networkmanager.conf", "/etc/bubba-networkmanager.conf" ) );
+    p.AddOption( Option( "pidfile", '\0', Option::String, "PID file", "/run/bubba-networkmanager.pid", "/run/bubba-networkmanager.pid" ) );
+    p.AddOption( Option( "socket", '\0', Option::String, "Socket to communicate on", "/run/bubba-networkmanager.sock", "/run/bubba-networkmanager.sock" ) );
+    p.AddOption( Option( "config", '\0', Option::String, "Configuration file to read from", "/etc/bubba/networkmanager.conf", "/etc/bubba/networkmanager.conf" ) );
 
     if ( !p.Parse( argc,argv ) ) {
         syslog( LOG_ERR,"Failed to parse arguments use %s -? for info",argv[0] );
--- scripts/client.py
+++ scripts/client.py
@@ -9,7 +9,7 @@
 
 def request(req):
 	client = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-	client.connect("/tmp/bubba-networkmanager.sock")
+	client.connect("/run/bubba-networkmanager.sock")
 	client.send(json.dumps(req))
 	d=client.recv(16384);
 	#print "%s"%d
--- utils/WlanCfg.cpp
+++ utils/WlanCfg.cpp
@@ -46,7 +46,7 @@
 using namespace EUtils;
 using namespace std;
 
-#define TZ_MAP_FILE "/usr/share/bubba-networkmanager/tz-lc.txt"
+#define TZ_MAP_FILE "/var/lib/bubba/tz-lc.txt"
 #define TZ_FILE "/etc/timezone"
 
 #if 0
